lib/colors.nix

```diff
@@ -1,65 +1,76 @@
 { lib, normalize-colors, ... }:
 let
-  /* Converts 2 digit hex to decimal number.
-
-     Example:
-       primaryHex2Dec "1a" = 26
-  */
-  primaryHex2Dec = hex:
+  inherit (builtins) getAttr;
+  compose = lib.flip lib.pipe;
+  base-digits = lib.stringToCharacters "0123456789abcdefghijklmnopqrstuvwxyz";
+  parseIntRadix =
+    base:
+    assert 2 <= base && base <= 36;
     let
-      hex2decDigits =
-        rec {
-          "0" = 0; "1" = 1; "2" = 2; "3" = 3; "4" = 4;
-          "5" = 5; "6" = 6; "7" = 7; "8" = 8; "9" = 9;
-          a = 10; b = 11; c = 12; d = 13; e = 14; f = 15;
-          A = a; B = b; C = c; D = d; E = e; F = f;
-        };
+      digits' = compose [
+        (lib.take base)
+        (lib.imap0 (lib.flip lib.nameValuePair))
+        lib.listToAttrs
+      ] base-digits;
     in
-      16 * hex2decDigits."${builtins.substring 0 1 hex}"
-      + hex2decDigits."${builtins.substring 1 2 hex}";
+    compose [
+      lib.toLower
+      lib.stringToCharacters
+      (map (lib.flip getAttr digits'))
+      (lib.foldr (cur: acc: acc * base + cur) 0)
+    ];
+  toIntBase16 = parseIntRadix 16;
```

Comment:
Does the `toIntBase16` function do the same as `primaryHex2Dec`? While the old name may not be the best, the old implementation is a lot simpler. Are there any benefits to your implementation to justify the complexity?

Reply:
`primaryHex2Dec` looks too specific; this implementation makes the intent more clear.

Reply:
What do you mean by clear and specific? Does it handle some existing usecase not handled by the old implementation? The code does not have any tests so I still prefer the old one as the only logic it has is 2 attrset lookups.

Reply:
I could add some tests if needed

Reply:
Overall, the library would benefit from tests, but this is one of the two cons. The second con is that I do not see how a composition of ~10 functions state the intent clearer than 2 attrset lookups.